### DISCUSSION TITLE
[Encryption] Addition of Data Encryption Feature

### DISCUSSION THREAD

**Discussion by DemonAngeX2** [immich-app/immich#2901] [Feature Request]
### The feature

Dear Development Team,

I hope this message finds you well. I am writing to request the addition of a crucial feature to the Immich application, specifically the implementation of data encryption for user photos and videos.

As a multi-user application that allows users to store their personal media on your server, it is essential to prioritize the security and privacy of their data. By introducing data encryption, you can provide users with an added layer of protection, ensuring that their photos and videos are securely stored and inaccessible to unauthorized individuals.

Implementing data encryption would involve encrypting user data before it is uploaded to the server and decrypting it when accessed by the authorized user. This approach would significantly enhance the confidentiality and integrity of the stored media, mitigating concerns regarding unauthorized access.

I understand that implementing such a feature may require significant development resources and considerations. However, it would be highly beneficial to both the users and the reputation of the Immich application. Data privacy and security are critical aspects of any modern application, and the addition of data encryption would position Immich as a trusted platform that prioritizes user privacy.

Thank you for considering my request. I believe that by implementing data encryption, Immich can provide a more secure and reliable service, fostering trust among its users. If there are any further details or discussions required, please do not hesitate to reach out.

Sincerely,
Alex

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by uhthomas** [comment]
Duplicate of #450.

**Comment by etnoy** [comment]
If you want encrypted data at rest you should encrypt your storage on the os level

**Reply by uhthomas** [reply]
https://github.com/immich-app/immich/issues/450#issuecomment-1595848717

**Comment by akostadinov** [comment]
From previous discussion

> all features except for viewing the photos would be gone

I think that these can be implemented client side, with some metadata be on server but not the actual unenecrypted images and videos.

For sharing, GPG exists since a long time.

I agree it is complicated and not the focus of the team which is alright. Just pointing out that it is not impossible.

**Reply by asudox** [reply]
Ente has on device AI models to recognize faces and classify images, it is not impossible.

And OP's text reads like some LLM generated text.

**Reply by bo0tzz** [reply]
We never claimed it's impossible. It just doesn't make any sense for Immich, since it would be a complete rewrite to an entirely different architecture with a bunch of tradeoffs.

**Comment by cakerack** [comment]
Hi - I have to say I am very close to using Immich but the main concern I have is the lack of any protection of data at rest.  Although the specific issue of a rogue admin isnt a concern for me (I would be the admin and the only user), the worry would be around the prospect of a zero-day and my data being compromised that way.  Encryption of data at the OS or FS level wouldn't mitigate that risk.  Just my thoughts...

**Reply by bo0tzz** [reply]
If you're worried about a vulnerability in Immich then don't expose it to the internet.

**Comment by tollsimy** [comment]
I think this would be a game-changer feature too, absolutely necessary imho and unfortunately what made me switch to another solution.

**Reply by etnoy** [reply]
Why don't you just consider full disk encryption?

**Reply by tollsimy** [reply]
I am already using it, but I won't see why the admin would be able to see other accounts data.

**Reply by etnoy** [reply]
Please see the above discussion, it's already been discussed why this can't be prevented

**Comment by DemonAngeX2** [comment]
Another app for cloud called pydio cells, got encryption for data uploaded from the webui. Maybe you can do the same thing ?

I'm not dev at all, i just give an idea ^^

**Reply by tollsimy** [reply]
Will give a look, thanks!

**Comment by bo0tzz** [comment]
This topic has been discussed to death in many threads; it's not going to happen.

**Reply by tollsimy** [reply]
that's unfortunate, thanks anyway

**Reply by akostadinov** [reply]
@tollsimy , think about it. How do you expect server side processing if you have client encryption? And if you want to sacrifice the features of server side processing to the encryption, then maybe use some other project. Somebody mentioned pydio which I hear for the first time. Maybe seafile, idk.

And if you want to do everything on the client side, things may easily go out of control. I heard that synology photos is doing just thumbnails on the client side and people have troubles with phones overheating. Not even talking about the possible network bottleneck. What's gonna happen if you add transcoding and machine learning?

**Reply by yesbatch** [reply]
@akostadinov I fully understand the position of the Immich authors. However, doing those things on the server is a design choice (an understandable one), not a requirement. If you want to compare to other products, an obvious one is Apple's. They have a photos product with similar (subjectively superior IMO) functionality without requiring that their servers access your media contents.

Also, personally I'm not running Immich on a powerful desktop. Most of my clients (even phone by most metrics) are more powerful than the NAS immich is running on.

**Reply by etnoy** [reply]
> @akostadinov I fully understand the position of the Immich authors. However, doing those things on the server is a design choice (an understandable one), not a requirement. If you want to compare to other products, an obvious one is Apple's. They have a photos product with similar (subjectively superior IMO) functionality without requiring that their servers access your media contents.
> 
> Also, personally I'm not running Immich on a powerful desktop. Most of my clients (even phone by most metrics) are more powerful than the NAS immich is running on.

Your comment makes zero sense

**Reply by bo0tzz** [reply]
The discussions about end-to-end encryption in Immich have all been played out many times already. There's nothing left to discuss, so I'll be locking this thread.
